src/controllers/orderController.js

```diff
@@ -0,0 +1,48 @@
+const { Order, OrderItem } = require('../models/Order');
+const { StatusCodes } = require('http-status-codes');
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
+const { BadRequestError } = require('../errors');
+
+const createOrder = async (req, res) => {
+  const { orderItems, subtotal, tax } = req.body;
+
+  try {
+    // Create an array of OrderItem objects
+    const orderItemObjects = orderItems.map(item => new OrderItem(item));
+    // Calculate the total price including tax
+    const total = subtotal + (subtotal * tax);
+
+    // Perform necessary checks and validations
+    if (orderItems.length === 0) {
+      throw new BadRequestError('Unable to create order because the order items are missing.'); 
+    }
+
+    // Create an Order document in the database
+    const order = await Order.create({
+      user: req.user.userId,
+      orderItems: orderItemObjects,
+      subtotal,
+      tax,
+      total,
+      orderStatus: 'pending', // Set initial order status as pending
+    });
+
+    // Create a payment intent with Stripe
+    const paymentIntent = await stripe.paymentIntents.create({
+      amount: total * 100, // Stripe requires amount in cents
+      currency: 'usd',
+      description: `Order #${order._id}`,
+    });
+
+    // Send the payment intent client secret and order information to the client
+    res.status(StatusCodes.CREATED).json({ clientSecret: paymentIntent.client_secret, order });
+  } catch (error) {
+    if (error instanceof BadRequestError) {
+      res.status(error.statusCode).json({ error: error.message });
+    } else {
+      res.status(StatusCodes.BAD_REQUEST).json({ error: error.message });
+    }
+  }
```

Comment:
Any reason to wrap this in a `try/catch` vs just letting the error handling middleware cover it? And could you explain why we want to return bad request 400 for _**all**_ error cases here?

Reply:
Wrapping this block in a try/catch is not necessary, I have removed it. Regarding returning a 400 Bad Request status for all errors, regardless of their type, I believed that it offers a clear signal to the client that something i
<img width="647" alt="screen3" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/d8d1cead-cbeb-40c1-a813-831e5b3a7b60">
s amiss with the request, but I agree, that was a strange idea. I'll adjust it to only return a 400 Bad Request status when the client sends a request with incorrect data.